### PR TITLE
assumptions

### PR DESCRIPTION
**Pull request by vlandham**

### CODE REVIEW

File: public/src/assumptions.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+// # Checking Data Assumptions
+//
+// Data processing is tricky business, full of pitfalls and gotchas. Hopefully the tasks in this guide help with getting started in this process. But you, I, and the entire world will make mistakes. Its natural.
```
Comment (iros) [inline public/src/assumptions.js]: its natural -> it's natural

File: src/assumptions.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  return true;
+}
+
+// This will output a given message if the input is not true. Typically assertions [throw](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw) errors, but we can just log it for explaining purposes.
+//
+// ## Data Content Assumptions
+//
+// Now let's use our `assert` function to check some assumptions about the details of our data.
+//
+// We can use lodash's suite of [type checking functions(https://lodash.com/docs#isBoolean) to take care of performing the checks, passing the result of the check to `assert` to produce our errors.
```
Comment (iros) [inline src/assumptions.js]: Broken link. Need closing square bracket after "type checking functions"

File: src/assumptions.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+];
+
+// Our first entry looks ok, where our second entry has some problems. The age parsing for the immortal [Sleepwalker](http://en.wikipedia.org/wiki/Sleepwalker_%28comics%29) has left him with no age. Also, bad input data has left us with a string in `superhuman`, where we expect a boolean. 
+//
+// A simple assumption checking function that could be run on this data could look something like this:
+
+function checkDataContent(data) {
+  data.forEach(function(d) {
+    var dString = JSON.stringify(d);
+    assert(_.isString(d.name), dString + " has a bad name ");
```
Comment (iros) [inline src/assumptions.js]: Does it make sense to make the messages indicate what the result should be? like "has a bad name. Should be a string"

**Comment by vlandham** [comment]
Awesome. thank you for all the catches.

**Pull request merged**